src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/hosts/hosts.component.ts

```diff
@@ -503,11 +494,13 @@ export class HostsComponent extends ListWithDetails implements OnDestroy, OnInit
         }),
         map((hostList: object[]) =>
           hostList.map((host) => {
+            const counts = {};
+            host['service_instances'] = new Set<string>();
+            host['services'].forEach((service: any) => {
```

Comment:
Looks good, but I think this data processing should be done in the backend to reduce load for frontend/browser and also we can cache the request in the backend or in general make processing more optimal.

Reply:
I fully agree with @avanthakkar . I didn't comment anything here because:
* This file already contained a lot of front-end logic (so it'd deserve a deeper refactoring),
* This work is for an environment where we have limited access, and changing things in the front-end only speeds up our fixes.

Reply:
Yup, even I agree with the backend processing. But for the reasons Ernesto said, I thought I'll just go with it. And I remember we already have a ticket to refactor the backend processing.